In [1]:
import pandas as pd
# uploading the first dataframe
df1 = pd.read_csv('df.csv')

In [2]:
df1

Location        Price       Bedrooms      Bathrooms        Toilets  \
0           Runda  100,000,000              5              5  Not available   
1      Kileleshwa   27,500,000              4              5              5   
2            Ruai    7,800,000              5              4              5   
3           Karen   75,500,000              5              5  Not available   
4           Haven  120,000,000              4              4              4   
..            ...          ...            ...            ...            ...   
315      Kilimani   26,000,000              3  Not available  Not available   
316      Kilimani   25,000,000  Not available  Not available  Not available   
317       Kihingo  165,000,000              4              5  Not available   
318  Chalbi Drive   66,000,000              4              5  Not available   
319   House No 21  120,000,000             10             10  Not available   

           Parking  
0    Not available  
1    Not available  
2                2  
3    Not available  
4                4  
..             ...  
315  Not available  
316  Not available  
317  Not available  
318  Not available  
319  Not available  

[320 rows x 6 columns]

This first dataframe has 320 rows and 6 columns

In [3]:
# uploading the second dataframe
df2 = pd.read_csv('stardata.csv')
df2

Location        Price  Bedrooms Bathrooms
0       Nairobi South, Starehe   26,000,000         4        3 
1               Karen, Langata  100,000,000         5        4 
2                      Nairobi   16,750,000         3        2 
3                      Nairobi   13,000,000         3        2 
4                      Nairobi   48,000,000         4        5 
...                        ...          ...       ...       ...
43051                  Nairobi   48,000,000         4        4 
43052                  Nairobi   65,000,000         4        4 
43053                  Nairobi   70,000,000         5        5 
43054                  Nairobi  400,000,000         4        5 
43055                  Nairobi   42,000,000         5        4 

[43056 rows x 4 columns]

The second datafrane has 43,056 rows and columns

In [4]:
df = pd.concat([df1, df2], axis=0)
df

Location        Price Bedrooms Bathrooms        Toilets  \
0           Runda  100,000,000        5         5  Not available   
1      Kileleshwa   27,500,000        4         5              5   
2            Ruai    7,800,000        5         4              5   
3           Karen   75,500,000        5         5  Not available   
4           Haven  120,000,000        4         4              4   
...           ...          ...      ...       ...            ...   
43051     Nairobi   48,000,000        4        4             NaN   
43052     Nairobi   65,000,000        4        4             NaN   
43053     Nairobi   70,000,000        5        5             NaN   
43054     Nairobi  400,000,000        4        5             NaN   
43055     Nairobi   42,000,000        5        4             NaN   

             Parking  
0      Not available  
1      Not available  
2                  2  
3      Not available  
4                  4  
...              ...  
43051            NaN  
43052            NaN  
43053            NaN  
43054            NaN  
43055            NaN  

[43376 rows x 6 columns]

df refers to the two dataframes combines and the dataframe has 43,376 rows and 6 columns. 

In [5]:
df['Location'].value_counts()

 Nairobi                   21527
 Karen, Langata            12558
 Nairobi South, Starehe     8970
Runda                         59
Lavington                     39
                           ...  
Ridge                          1
Hardy Area                     1
Mzima Springs                  1
Muthaiga North Rd              1
Nairobi                        1
Name: Location, Length: 108, dtype: int64

In [6]:
df['Bedrooms'].value_counts()

5                23322
4                14352
3                 5382
5                  139
4                  104
3                   26
6                   25
7                    9
8                    5
10                   4
14                   2
2                    2
Not available        2
16                   1
9                    1
Name: Bedrooms, dtype: int64

In [7]:
df['Bathrooms'].value_counts()

5                16146
4                12558
3                 5382
2                 5382
7                 1794
House             1794
5                   93
6                   82
4                   46
7                   26
Not available       23
3                   19
2                    8
9                    6
8                    6
10                   4
12                   1
340                  1
180                  1
15                   1
11                   1
1                    1
18                   1
Name: Bathrooms, dtype: int64

In [8]:
df['Toilets'].value_counts()

6                89
5                73
Not available    46
4                35
7                27
3                16
2                 9
1                 8
9                 5
8                 4
10                3
12                1
18                1
15                1
210               1
11                1
Name: Toilets, dtype: int64

In [9]:
df['Toilets'].value_counts()

6                89
5                73
Not available    46
4                35
7                27
3                16
2                 9
1                 8
9                 5
8                 4
10                3
12                1
18                1
15                1
210               1
11                1
Name: Toilets, dtype: int64

In [10]:
df['Parking'].value_counts()

Not available    103
4                 43
5                 35
6                 30
3                 28
10                21
2                 18
8                 14
7                 11
9                  5
15                 3
20                 3
14                 1
4,830              1
30                 1
5,000              1
2,023              1
1                  1
Name: Parking, dtype: int64

In [11]:
class DataUnderstanding(object):
    """This is a class that does basic data understanding"""
    def __init__(self, df):
        self.shape = df.shape
        self.info = df.info
        self.duplicates = df.duplicated().sum()
        self.missing = df.isna().sum()
        self.types = df.dtypes

In [12]:
#instantiating the classes
dataund= DataUnderstanding(df)

In [13]:
#Getting the shape of the dataset
print(f"shape:{dataund.shape}")

shape:(43376, 6)


In [14]:
#getting the info of the data
print(dataund.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 43376 entries, 0 to 43055
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Location   43376 non-null  object
 1   Price      43376 non-null  object
 2   Bedrooms   43376 non-null  object
 3   Bathrooms  43376 non-null  object
 4   Toilets    320 non-null    object
 5   Parking    320 non-null    object
dtypes: object(6)
memory usage: 2.3+ MB
None


In [15]:
#getting the missing values
dataund.missing

Location         0
Price            0
Bedrooms         0
Bathrooms        0
Toilets      43056
Parking      43056
dtype: int64

In [16]:
#checking the data types
dataund.types

Location     object
Price        object
Bedrooms     object
Bathrooms    object
Toilets      object
Parking      object
dtype: object

In [17]:
#checking for duplicates
print(f"Duplictes:{dataund.duplicates.sum()}")

Duplictes:43048


## Data Preparation 